sw/device/lib/testing/mock_mmio_test.cc

```diff
@@ -24,13 +24,46 @@ uint32_t WriteTwice(mmio_region_t dev) {
   return value;
 }
 
-class WriteTwiceTest : public Test, public MmioTest {};
-TEST_F(WriteTwiceTest, WriteTwice) {
+class MockMmioTest : public Test, public MmioTest {};
+
+TEST_F(MockMmioTest, WriteTwice) {
   EXPECT_READ32(0x0, 0xdeadbeef);
   EXPECT_WRITE32(0x4, 0x21524110)
   EXPECT_WRITE16(0x8, 0xdead);
   EXPECT_WRITE16(0xa, 0xbeef);
 
   EXPECT_EQ(WriteTwice(dev().region()), 0xdeadbeef);
 }
+
+TEST_F(MockMmioTest, ExpectWithString) {
+  EXPECT_READ32(0xc, "*\0\0*");
+  EXPECT_EQ(mmio_region_read32(dev().region(), 0xc), 0x2a00002a);
+
+  EXPECT_WRITE32(0x12, "abcd");
```

Comment:
Hmmm, is this right? As in the example you set of expectation of 5 bytes (`"abcd"` is `'a'`, `'b'`, `'c'`, `'d'`, `'\0'`), and the you write a 4 byte value.

I think maybe this have to be separate to `EXPECT_WRITE/EXPECT_READ` macro, similar to the `EXPECT_MASK` macro?

I think  in terms of UART this feature is probably not so useful, but could be more useful for something like HMAC, where FIFOs support word size reads/writes.